src/analyze/analyze-time-data.c

```diff
@@ -254,14 +287,15 @@ int acquire_time_data(sd_bus *bus, bool require_finished, UnitTimes **out) {
                 return bus_log_parse_error(r);
 
         while ((r = bus_parse_unit_info(reply, &u)) > 0) {
-                UnitTimes *t;
+                _cleanup_(unit_times_donep) UnitTimes *t = NULL;
 
                 if (!GREEDY_REALLOC(unit_times, c + 2))
                         return log_oom();
 
                 unit_times[c + 1].has_data = false;
                 t = &unit_times[c];
                 t->name = NULL;
+                t->related = (RelatedUnits){ };
```

Comment:
I think `GREEDY_REALLOC0()` is better.

Reply:
I initially thought this suggestion was incorrect, but I think you're right: the clear() function would initialize the item in case of reuse. So, applying it.